src/Kiota.Builder/CodeRenderers/CodeRenderer.cs

```diff
@@ -21,13 +21,16 @@ public CodeRenderer(GenerationConfiguration configuration)
         _configuration = configuration;
         _rendererElementComparer = configuration.ShouldRenderMethodsOutsideOfClasses ? new CodeElementOrderComparerWithExternalMethods() : new CodeElementOrderComparer();
     }
-    public async Task RenderCodeNamespaceToSingleFileAsync(LanguageWriter writer, CodeElement codeElement, string outputFile, CancellationToken cancellationToken)
+    public async Task RenderCodeNamespaceToSingleFileAsync(LanguageWriter writer, string outputFile, CancellationToken cancellationToken, params CodeElement[] codeElements)
     {
         await using var stream = new FileStream(outputFile, FileMode.Create);
 
         var sw = new StreamWriter(stream);
         writer.SetTextWriter(sw);
-        RenderCode(writer, codeElement);
+
+        foreach (var codeElement in codeElements)
+            RenderCode(writer, codeElement);
```

Comment:
I'm not sure why we still need that change anymore since now CodeFile handles the aggregation?